js/common/modules/@arangodb/test-helper-common.js

```diff
@@ -362,3 +362,22 @@ exports.endpointToURL = (endpoint) => {
       return 'http' + endpoint.substr(pos);
   }
 };
+
+exports.checkIndexMetrics = (checkFunction) => {
+  var isMetricsArrived = false;
+  var timeToWait = 100;
+  while (!isMetricsArrived) {
+    try {
+      checkFunction();
+      isMetricsArrived = true;
+    } catch (err) {
+      isMetricsArrived = false;
+      if (timeToWait > 0) {
+        require("internal").sleep(2);
+        timeToWait = timeToWait - 2;
+      } else {
+        assertTrue(false);
```

Comment:
That should save us initial call stack and error
```suggestion
        throw err;
```